multimodal/src/autogluon/multimodal/models/sam.py

```diff
@@ -0,0 +1,193 @@
+    """
+
+    def __init__(
+        self,
+        prefix: str,
+        checkpoint_name: str,
+        config: DictConfig,
+        pretrained: Optional[bool] = True,
+        frozen_layers: Optional[list] = None,
+    ):
+        """
+        Load a pretrained Segment Anything Model (SAM).
+
+        Parameters
+        ----------
+        prefix
+            The prefix of the SAMForSemanticSegmentation model.
+        checkpoint_name
+            Name of the SAM checkpoint.
+        pretrained
+            Whether using the pretrained SAM models. If pretrained=True, download the pretrained model.
+        frozen_layers
+            A list of substrings of frozen layers' names.
+        """
+
+        super().__init__()
+        self.prefix = prefix
+        self.pretrained = pretrained
+        self.checkpoint_name = checkpoint_name
+        self.frozen_layers = frozen_layers
+
+        self.device = None
+        self.name_to_id = {}
+
+        self._load_checkpoint(checkpoint_name)
+
+        freeze_model_layers(self.model, self.frozen_layers)
+
+        self.image_size = self.model.vision_encoder.image_size
+
+        # for Binary Semantic Segmentation Tasks.
+        self.model.mask_decoder.num_mask_tokens = 1
+        mask_token_data = self.model.mask_decoder.mask_tokens.weight.data[0]
+        self.model.mask_decoder.mask_tokens = nn.Embedding(1, self.model.mask_decoder.hidden_size)
+        self.model.mask_decoder.mask_tokens.weight.data[0] = mask_token_data
+        hyper_mlps = self.model.mask_decoder.output_hypernetworks_mlps[0]
+        self.model.mask_decoder.output_hypernetworks_mlps = nn.ModuleList([hyper_mlps])
+
+        self.config = self.model.config
+
+    def _load_checkpoint(self, checkpoint_name):
+        if self.pretrained:
+            self.model = SamModel.from_pretrained(checkpoint_name)
+        else:
+            configuration = SamConfig(name_or_path=checkpoint_name)
+            self.model = SamModel(configuration)
+
+    def set_data_preprocessor_device(self):
+        if not self.device:
+            self.device = next(self.model.parameters()).device
+        if self.device != self.data_preprocessor.device:
+            self.data_preprocessor.to(self.device)
+
+    def save(self, save_path: str = "./", tokenizers: Optional[dict] = None):
+        weights_save_path = os.path.join(save_path, "model.pth")
+        configs_save_path = os.path.join(save_path, "config.py")
+
+        self._save_weights(save_path=weights_save_path)
+        self._save_configs(save_path=configs_save_path)
+
+        return save_path
+
+    def _save_weights(self, save_path=None):
+        if not save_path:
+            save_path = f"./{self.checkpoint_name}_autogluon.pth"
+
+        torch.save({"state_dict": self.model.state_dict()}, save_path)
+
+    def _save_configs(self, save_path=None):
+        if not save_path:
+            save_path = f"./{self.checkpoint_name}_autogluon.py"
+
+        self.config.dump(save_path)
```

Comment:
Consider using huggingface's API to save the model. Looks like no need to save config? Maybe remove `_save_weights` and `_save_configs` and only keep `save`.